src/sentry/grouping/strategies/base.py

```diff
@@ -127,7 +127,7 @@ def get_grouping_component(
 
         if self["variant"] is not None:
             assert len(rv) == 1
-            return rv[self["variant"]]
+            return {"only_one_variant": rv[self["variant"]]}
```

Comment:
This is a weird trick I'm using to help this function be of the same return type. I'm open to suggestions.

This codebase seems to be covered:
<img width="452" alt="image" src="https://github.com/getsentry/sentry/assets/44410/cf7abf6a-7302-48ff-a441-5aba58f57f54">